lib/flappi/api_doc_formatter.rb

```diff
@@ -22,7 +22,7 @@ def self.format_to_text(doc)
 
       template_path = "#{Pathname.new(method(:format).source_location.first).dirname}/api_doc_template.rb.erb"
       # NOTE: this will warn but will fail if fixed as suggested in Ruby docs
-      template = ERB.new File.read(template_path), nil, '-'
+      template = ERB.new(File.read(template_path), trim_mode: '-')
```

Comment:
Is the comment still relevant above hector? If not perhaps we can remove it so its not misleading

Reply:
Oh shit, ye that doc looks dangerous 😢

Reply:
OK, thanks for bringing this to my attention. It really is a warning sign.

So Richard in fdc6a45ef5e73d0661b7927b347ad11010ae670c fixed the warning as Ruby suggested. Then in 56c2e100be342b00828468a7462bc7f422af2c30 he changed it back, and added this comment "# NOTE: this will warn but will fail if fixed as suggested in Ruby docs".

I'm not sure why. But I don't believe that Ruby doc is really wrong. I have examined the source code, and I believe the following two ways are the same.

```diff
- template = ERB.new File.read(template_path), nil, '-'
+ template = ERB.new(File.read(template_path), trim_mode: '-')
```

Also there are multiple tests that cover this code, and they pass.

So I have my suspicions. How about we put this on testing and see if it breaks? I'll leave this discussion open so that depending on how testing goes I'll remember to come back and clean up the comment or change the code back.

Reply:
@FunnyHector 👍

Reply:
API pages on testing look exactly the same to me. Let's see how QA see about it.

Reply:
So Sophie has passed QA (see Vimaly), I think it's safe to ship it. I trust Sophie 100% 😬